packages/react/src/lib/applyAccessibilityKeyHandlers.ts

```diff
@@ -0,0 +1,24 @@
+import * as _ from 'lodash'
+
+import { Props, ShorthandValue } from '../types'
+import { KeyboardHandler, OnKeyDownHandler } from './accessibility/types'
+
+const applyAccessibilityKeyHandlers = (
+  keyHandlers: OnKeyDownHandler,
+  value: Props | ShorthandValue,
+) => {
+  const slotProps: Props = typeof value === 'object' ? value : {}
```

Comment:
Let's not ignore the other cases of shorthand value. Let's refactor this in the same way it is done in the factories, something like:
```  
  const valIsPropsObject = _.isPlainObject(value)
  const valIsReactElement = React.isValidElement(value)

  const slotProps =
    (valIsReactElement && (value as React.ReactElement<Props>).props) ||
    (valIsPropsObject && (value as Props)) ||
    {}
```